src/lib/access-control/page-permission-helper.ts

```diff
@@ -31,7 +33,12 @@ export const getProperPermissionType = (permissionA: PagePermissionType = PAGE_P
 };
 
 const menuIdList = Object.keys(MENU_INFO_MAP) as MenuId[];
-export const getPermissionRequiredMenuIds = (): MenuId[] => menuIdList.filter((id) => MENU_INFO_MAP[id]?.needPermissionByRole);
+export const getPermissionRequiredMenuIds = (): MenuId[] => menuIdList.filter((id) => {
+    const menu = MENU_INFO_MAP[id];
+    if (!menu) return false;
+    if (id === MENU_ID.DASHBOARDS && config.get('DASHBOARD_ENABLED')) return menu.needPermissionByRole;
+    return menu.needPermissionByRole;
```

Comment:
question: aren't these two codes the same?

```suggestion
    return menu.needPermissionByRole;
```